### PR TITLE
Updated optional dependencies to 3.9

### PR DESCRIPTION
**Pull request by oscargus**
(and removed 2.7-tests)

#### References to other Issues or PRs

Seeing if I can get optional dependencies on Travis to work by upgrading Python version.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -27,13 +27,11 @@ jobs:
...
         - TEST_SYMPY="true" SPLIT="2/2"
-    - python: 3.7
+    - python: 3.9
       dist: xenial
-      # At the time of writing this is Python 3.7 but it will become 3.8 when
-      # all of the dependencies are supported on 3.8.
       env:
         - TEST_ASCII="true"
-        - TEST_OPT_DEPENDENCY="matchpy numpy scipy gmpy2 matplotlib aesara llvmlite autowrap cython wurlitzer python-symengine tensorflow numexpr ipython antlr-python-runtime>=4.7,<4.8 antlr>=4.7,<4.8 cloudpickle pyglet pycosat lfortran python-clang lxml"
+        - TEST_OPT_DEPENDENCY="matchpy numpy scipy gmpy2 matplotlib aesara autowrap cython wurlitzer python-symengine tensorflow numexpr ipython antlr-python-runtime>=4.7,<4.8 antlr>=4.7,<4.8 cloudpickle pycosat lxml" # llvmlite lfortran python-clang pyglet
```
Comment (asmeurer) [inline .travis.yml:34]: These aren't installable with 3.9 yet? What about 3.8?

Comment (oscargus) [reply .travis.yml:34]: Not sure what the problem is. There is a 800 MB download including openjdk sp I tried to remove some packages to see if the size was reduced. (It was, but 9nly to 700 MB)

If I understand it correctly it cannot install that much and dies. However,  right now it is not the optional dependencies that fails...

Comment (asmeurer) [reply .travis.yml:34]: https://github.com/sympy/sympy/issues/21764#issuecomment-1006078052

Comment (asmeurer) [reply .travis.yml:34]:
```suggestion
        - TEST_OPT_DEPENDENCY="matchpy numpy scipy gmpy2 matplotlib aesara llvmlite autowrap cython wurlitzer python-symengine tensorflow numexpr ipython antlr-python-runtime>=4.7,<4.8 cloudpickle pyglet pycosat lfortran python-clang lxml"
```

Comment (isuruf) [reply .travis.yml:34]: In conda-forge `matplotlib = maptlotlib-base + pyqt` which means we can use `matplotlib-base` if we don't need the pyqt backend.

File: .travis.yml [hunk 2]
```diff
@@ -27,13 +27,11 @@ jobs:
...
         - TEST_SYMPY="true" SPLIT="2/2"
-    - python: 3.7
+    - python: 3.9
       dist: xenial
-      # At the time of writing this is Python 3.7 but it will become 3.8 when
-      # all of the dependencies are supported on 3.8.
       env:
         - TEST_ASCII="true"
-        - TEST_OPT_DEPENDENCY="matchpy numpy scipy gmpy2 matplotlib aesara llvmlite autowrap cython wurlitzer python-symengine tensorflow numexpr ipython antlr-python-runtime>=4.7,<4.8 antlr>=4.7,<4.8 cloudpickle pyglet pycosat lfortran python-clang lxml"
+        - TEST_OPT_DEPENDENCY="matchpy numpy scipy gmpy2 matplotlib aesara llvmlite autowrap cython wurlitzer python-symengine tensorflow numexpr ipython antlr-python-runtime>=4.7,<4.8 cloudpickle pyglet pycosat lfortran python-clang lxml"
```
Comment (isuruf) [inline .travis.yml:34]:
```suggestion
        - TEST_OPT_DEPENDENCY="matchpy numpy scipy gmpy2 matplotlib-base aesara llvmlite autowrap cython wurlitzer python-symengine tensorflow numexpr ipython antlr-python-runtime>=4.7,<4.8 cloudpickle pyglet pycosat lfortran python-clang lxml"
```

**Pull request closed without merging**